src/main/java/hudson/plugins/git/extensions/impl/CloneOption.java

```diff
@@ -148,7 +149,20 @@ public void decorateCloneCommand(GitSCM scm, Run<?, ?> build, GitClient git, Tas
             // in a single job definition.
             RemoteConfig rc = scm.getRepositories().get(0);
             List<RefSpec> refspecs = rc.getFetchRefSpecs();
-            cmd.refspecs(refspecs);
+            List<RefSpec> expandedrefSpecs = new ArrayList<>();
+            Boolean check = false;
+            for (RefSpec ref:refspecs) {
+                if(ref.toString().contains("$")){
```

Comment:
If one `ref` doesn't contain `$`, it won't be added to `expandedrefSpecs`. However, it another `ref` contains `$`, `expandedrefSpecs` would still be used. Is that intentional?

Reply:
No it wasn't intentional, this logic wouldn't cover all the cases, thanks for pointing that out!